go.mod

```diff
@@ -1,9 +1,12 @@
 module github.com/containerd/nri
 
-go 1.22.0
+go 1.23.0
+
+toolchain go1.23.4
```

Comment:
Do we need this update as part of this PR?

Reply:
Hmm... looks like we don't need it. With all the golang bumps reverted, everything still builds, golangci-lint is clean and all tests pass. Weird... I could swear at one point I had to bump. Anyway, thanks for questioning. Reverted the version bumps.